src/passes/referencedLibraries.ts

```diff
@@ -37,46 +31,64 @@ export class ReferencedLibraries extends ASTMapper {
         return this.visitExpression(node, ast);
       }
 
-      //Checks if the Function is a referenced Library functions,
-      //if yes add it to the linearizedBaseContract list of parent ContractDefinition node
-      //free functions calling library functions are not yet supported
-      librariesById.forEach((library, _) => {
-        if (library.vFunctions.some((libraryFunc) => libraryFunc.id === calledDeclaration.id)) {
-          const parent = node.getClosestParentByType(ContractDefinition);
-          if (parent === undefined) return;
+      // Free functions calling library functions are not yet supported
+      const parent = node.getClosestParentByType(ContractDefinition);
+      if (parent === undefined) return;
```

Comment:
In that case, we should halt the transpilation with proper msg rather than ignoring it.

Reply:
The previous pass is in charge of adding the free functions to the contracts, therefore there is no problem with those functions that referenced libraries since in this pass they are going to be treated as functions of the contract. Since free functions are not removed from the ast, they should be ignored in this case.